src/workerd/io/worker-interface.capnp

```diff
@@ -157,6 +157,125 @@ struct Trace @0x8e8d911203762d34 {
   executionModel @25 :ExecutionModel;
   # the execution model of the worker being traced. Can be stateless for a regular worker,
   # durableObject for a DO worker or workflow for the upcoming Workflows feature.
+
+  # =====================================================================================
+  # Additional types for streaming tail workers
+
+  struct Attribute {
+    # An Attribute mark is used to add detail to a span over its lifetime.
+    # The Attribute struct can also be used to provide arbitrary additional
+    # properties for some other structs.
+    # Modeled after https://opentelemetry.io/docs/concepts/signals/traces/#attributes
+    struct Value {
+      inner :union {
+        text @0 :Text;
+        bool @1 :Bool;
+        int @2 :Int32;
+        float @3 :Float64;
+      }
+    }
+    name @0 :Text;
+    value @1 :List(Value);
+  }
+
+  struct Return {
```

Comment:
Are return events supposed to be issued for every span? A lot of the time there won't be any meaningful time between the return and the span close right?

Reply:
No, not for every span, only for cases where it makes sense. For instance, with a fetch subrequest like the following... the span open will correspond with the fetch request itself, the return marks the moment the fetch response was received, but the span close won't come until after the full response payload is read. Other types of spans may not have any kind of meaningful return and therefore wouldn't ever have a `Return` mark.